src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/BotSegmentsPlaybackController.cs

```diff
@@ -486,6 +490,7 @@ private void EvaluateBotSegments()
                         nextBotSegmentIndex == 0,
                         nextBotSegment.Replay_SegmentNumber,
                         nextBotSegment.Replay_ActionCompleted,
+                        nextBotSegment.Replay_ValidationsCompleted,
                         nextBotSegment.endCriteria
                     );
 
```

Comment:
about line 520 in your new code.. there is logic for waiting for actions to complete even if the user didn't specify the end criteria to wait for actions to complete....

I feel like this should follow a similar pattern...

After the action has stopped AND we're in that hasMatched endCriteria section...
we then would tell the validation to stop.  At this point, any needed criteria that should have matched and haven't would fail.  However, this would solve you issue where you have things that should 'never' be a condition.. at this point you would mark them as passed if they were still correct.

Reply:
Perfect, that is exactly what I was looking for.

Does this comment mean that I should _only_ do this? What did you think of the "ValidationsCompleted" endCriteria type? I could see a case for both.

Reply:
Ok I think I got this working in a pretty decent way, was a little weird at first. There are two options:

1. If you add an endCriteria of ValidationsCompleted, it will wait for the validations to be set before ending the segment. There is also a timeout now so you can wait a certain amount of time and then end the segment anyways (this does not stop the next segment from moving on)
2. OR, if you don't provide a ValidationsCompleted criteria, it will automatically stop the validations as soon as all criteria and actions have been met. Technically you can still include a timeout, but I don't think that would be used much in this situation.